lib/vfscore/lookup.c

```diff
@@ -223,21 +227,37 @@ namei(const char *path, struct dentry **dpp)
 				node[0] = 0;
 
 				if (++links_followed >= MAXSYMLINKS) {
-					return (ELOOP);
+					error = ELOOP;
+					goto out;
 				}
 				need_continue = 1;
 				break;
 			}
 
 			if (*p == '/' && ddp->d_vnode->v_type != VDIR) {
 				drele(ddp);
-				return ENOTDIR;
+				error = ENOTDIR;
+				goto out;
 			}
 		}
 	} while (need_continue);
 
 	*dpp = dp;
-	return 0;
+out:
+	if (realpath)
+		strcpy(realpath, fp);
```

Comment:
```suggestion
		strlcpy(realpath, fp, PATH_MAX);
```

We can use `strlcpy()` in order to keep the consistency [within the file](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/lookup.c#L87) and for better security.